src/function/scalar/list/list_resize.cpp

```diff
@@ -0,0 +1,253 @@
+	switch (child_type) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		TemplatedListResizeFunction<int8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT16:
+		TemplatedListResizeFunction<int16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT32:
+		TemplatedListResizeFunction<int32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT64:
+		TemplatedListResizeFunction<int64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::INT128:
+		TemplatedListResizeFunction<hugeint_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT8:
+		TemplatedListResizeFunction<uint8_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT16:
+		TemplatedListResizeFunction<uint16_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT32:
+		TemplatedListResizeFunction<uint32_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::UINT64:
+		TemplatedListResizeFunction<uint64_t, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::FLOAT:
+		TemplatedListResizeFunction<float, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::DOUBLE:
+		TemplatedListResizeFunction<double, StandardCopyValue>(args, result);
+		break;
+	case PhysicalType::VARCHAR:
+		TemplatedListResizeFunction<string_t, StringCopyValue>(args, result);
+		break;
+	case PhysicalType::INTERVAL:
+		TemplatedListResizeFunction<interval_t, NestedCopyValue>(args, result);
+		break;
+	case PhysicalType::STRUCT:
+	case PhysicalType::LIST:
+		TemplatedListResizeFunction<int8_t, NestedCopyValue>(args, result);
+		break;
+	default:
+		throw NotImplementedException("This function has not been implemented for logical type %s",
+		                              TypeIdToString(child_type));
+	}
+}
+
+static unique_ptr<FunctionData> ListResizeBind(ClientContext &context, ScalarFunction &bound_function,
+                                               vector<unique_ptr<Expression>> &arguments) {
+	D_ASSERT(bound_function.arguments.size() == 2 || arguments.size() == 3);
+	bound_function.arguments[1] = LogicalType::UBIGINT;
+
+	// first argument is constant NULL
+	if (arguments[0]->return_type == LogicalType::SQLNULL) {
+		bound_function.arguments[0] = LogicalType::SQLNULL;
+		bound_function.return_type = LogicalType::SQLNULL;
+		return make_uniq<VariableReturnBindData>(bound_function.return_type);
+	}
+
+	// prepared statements
+	if (arguments[0]->return_type == LogicalType::UNKNOWN) {
+		bound_function.return_type = arguments[0]->return_type;
+		return make_uniq<VariableReturnBindData>(bound_function.return_type);
+	}
+
+	// default type does not match list type
+	if (bound_function.arguments.size() == 3 &&
+	    ListType::GetChildType(arguments[0]->return_type) != arguments[2]->return_type &&
+	    arguments[2]->return_type != LogicalTypeId::SQLNULL) {
+		throw InvalidInputException("Default value must be of the same type as the lists or NULL");
```

Comment:
Could we insert a cast here instead by setting `bound_function.arguments[2]->return_type = arguments[0]->return_type`? That should allow "fuzzy mismatches", e.g.:

```sql
select list_resize([1, 2, 3]::BIGINT[], 5, 42);
```